compiler/rustc_hir_analysis/src/collect/resolve_bound_vars.rs

```diff
@@ -114,7 +114,7 @@ enum Scope<'a> {
         /// If this binder comes from a where clause, specify how it was created.
         /// This is used to diagnose inaccessible lifetimes in APIT:
         /// ```ignore (illustrative)
-        /// fn foo(x: impl for<'a> Trait<'a, Assoc = impl Copy + 'a>) {}
+        /// fn foo(x: impl for<'a> Trait<'a, Assoc = impl Copy >) {}
```

Comment:
Could also mark the example as `edition2024,ignore` to make clear that the lifetime is captured.  Otherwise, or to be explicit, the `+ use<'a>` you suggest makes more sense to me.